runtime/integration-tests/src/generic/envs/runtime_env.rs

```diff
@@ -82,14 +83,24 @@ impl<T: Runtime> Env<T> for RuntimeEnv<T> {
 		.assimilate_storage(&mut parachain_storage)
 		.unwrap();
 
+		// Precompiles need to have code-set
+		pallet_evm::GenesisConfig::<T> {
+			accounts: runtime_common::evm::precompile::utils::precompile_account_genesis::<
+				centrifuge_runtime::CentrifugePrecompiles,
```

Comment:
This should be:

```suggestion
				T::PrecompilesType
```

Although I think the correct place for this is `utils/genesis.rs` to use it only for those tests that need `pallet_evm`.

Reply:
No, that is a general setup. Took us 3 weeks to discover it again, we had it already discovered 6 month ago. So this should be default for all setups.

Reply:
`T::RecompilesType` does not work 
<img width="1097" alt="image" src="https://github.com/centrifuge/centrifuge-chain/assets/28841470/a41f8140-b118-4ddc-8e87-77369e89f410">

I do not want to carry that to the `trait Runtime` as a where clause, this will need to be carried to every location otherwise IIRC.

Reply:
Fixed it here: [125bc47](https://github.com/centrifuge/centrifuge-chain/pull/1720/commits/125bc47dff38b6355b73b107ec5040501a845e37)

Reply:
Ok, if we only use it there, we can leave it as it is.

In case we need to use it in more places, then you can add a new associated type to `trait Runtime` as `type PrecompilesTypeExt: H160Addresses` and then instead of adding the pallet as `pallet_evm::Config`, adding it as: `pallet_evm::Config<PrecompileType = Self::PrecompileTypeExt>`

Later in the above code you can use `T::PrecompileTypeExt`

Reply:
Yeah, thought about that too, but seemed to much work for the one location. Maybe will do later.

Reply:
Not mandatory at all!